packages/default-storage/android/build.gradle

```diff
@@ -1,87 +1,38 @@
-        if (candidatePath.toFile().exists()) {
-            return candidatePath.toString()
-        }
-
-        basePath = basePath.getParent()
-    }
-
-    return null
-}
-
-def safeExtGet(prop, fallback) {
-    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
-}
-
-def getFlagOrDefault(flagName, defaultValue) {
-    rootProject.hasProperty(flagName) ? rootProject.properties[flagName] == "true" : defaultValue
-}
-
-def getVersionOrDefault(String flagName, String defaultVersion) {
-    rootProject.hasProperty(flagName) ? rootProject.properties[flagName] : defaultVersion
-}
-
-def isNewArchitectureEnabled() {
-    // To opt-in for the New Architecture, you can either:
-    // - Set `newArchEnabled` to true inside the `gradle.properties` file
-    // - Invoke gradle with `-newArchEnabled=true`
-    // - Set an environment variable `ORG_GRADLE_PROJECT_newArchEnabled=true`
-    return project.hasProperty("newArchEnabled") && project.newArchEnabled == "true"
-}
-
 configurations {
     compileClasspath
 }
 
 buildscript {
-    // kotlin version is dictated by rootProject extension or property in gradle.properties
-    ext.asyncStorageKtVersion = rootProject.ext.has('kotlinVersion')
-            ? rootProject.ext['kotlinVersion']
-            : rootProject.hasProperty('AsyncStorage_kotlinVersion')
-            ? rootProject.properties['AsyncStorage_kotlinVersion']
-            : '1.8.10'
+    apply from: "config.gradle"
+    def kotlinVersion = ext.AsyncStorageConfig.kotlinVersion
+    def kspVersion = ext.AsyncStorageConfig.kspVersion
 
     repositories {
         mavenCentral()
         google()
     }
     dependencies {
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$asyncStorageKtVersion"
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion"
+        classpath "com.google.devtools.ksp:symbol-processing-gradle-plugin:$kspVersion"
     }
 }
 
-// AsyncStorage has default size of 6MB.
-// This is a sane limit to protect the user from the app storing too much data in the database.
-// This also protects the database from filling up the disk cache and becoming malformed.
-// If you really need bigger size, please keep in mind the potential consequences.
-long dbSizeInMB = 6L
-def newDbSize = rootProject.properties['AsyncStorage_db_size_in_MB']
-if (newDbSize != null && newDbSize.isLong()) {
-    dbSizeInMB = newDbSize.toLong()
-}
 
-// Instead of reusing AsyncTask thread pool, AsyncStorage can use its own executor
-def useDedicatedExecutor = getFlagOrDefault('AsyncStorage_dedicatedExecutor', false)
+apply plugin: 'com.android.library'
+apply from: "config.gradle"
 
-// Use next storage implementation
-def useNextStorage = getFlagOrDefault("AsyncStorage_useNextStorage", false)
+boolean isNewArchitectureEnabled = ext.AsyncStorageConfig.isNewArchitectureEnabled
+boolean useNextStorage = ext.AsyncStorageConfig.useNextStorage
+
+println("[AsyncStorage] Config: ${ext.AsyncStorageConfig}")
```

Comment:
Do we intend to keep this?

Reply:
Removed it, but I think it might come useful for debugging purposes if we keep it. WDYT?

Reply:
That makes sense. Can we use [`logger.info`](https://docs.gradle.org/current/userguide/logging.html#sec:sending_your_own_log_messages) to log this info?

Reply:
👍